heroku/cli
# refactor(redis-v5): Move `redis:wait` command to CLI

## Description

Moves the `redis:wait` command to CLI, converting to OClif Core and TS.

## Test instructions

You can test by pulling this branch, installing dependencies with `yarn` and building with `yarn build`. Then run the following commands, knowing that if the app doesn't have any Redis add-on, you should see an error and in apps with more than one Redis add-on you should see an error unless you specify the DATABASE arg.

```
./bin/run redis:wait --help                                    # always test that help looks ok.
./bin/run redis:wait -a test_app_without_redis                 # shows error.
./bin/run redis:wait -a test_app_single_redis                  # waits if there's a pending action.
./bin/run redis:wait -a test_app_multiple_redis                # shows error.
./bin/run redis:wait addon-name -a test_app_multiple_redis     # waits if there's a pending action.
```

The only way I found to make the command to actually wait on something was by issuing the command right after you add a new Heroku Redis add-on to your test app (and passing the add-on name on the command if there's more than one). The message returned by Redis API in that case is `preparing`, and you should see that written after the spinner, until it's done.

You can add Heroku Redis add-ons to any test app with `heroku addons:create heroku-redis -a test-app`.

## SOC2 Compliance
[GUS Work Item](https://gus.lightning.force.com/a07EE00001axf57YAA)
